pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -145,58 +153,42 @@ func (r *Reconciler) ensureGlobalPullSecret(ctx context.Context, operatorSecret,
 		return nil, errors.New("nil operator secret, cannot verify userData integrity")
 	}
 
-	recreate := false
-
-	// if there is no userSecret, create new, or when
-	// userSecret have broken type, recreates it with proper type
-	// unfortunately the type field is immutable, therefore the whole secret have to be deleted and create once more
-	if userSecret == nil || (userSecret.Type != corev1.SecretTypeDockerConfigJson || userSecret.Data == nil) {
-		recreate = true
-	}
-
-	if recreate {
-		secret = &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      pullSecretName.Name,
-				Namespace: pullSecretName.Namespace,
-			},
-			Type: corev1.SecretTypeDockerConfigJson,
-			Data: make(map[string][]byte),
+	// recreate := false
+	secretData := make(map[string][]byte)
+
+	var invalidSecretType bool
+	if userSecret != nil {
+		// when userSecret has broken type, recreates it with proper type
+		// unfortunately the type field is immutable, therefore the whole secret has to be deleted and recreated
+		if userSecret.Type != corev1.SecretTypeDockerConfigJson {
+			invalidSecretType = true
+			err := r.secretsClient.Delete(ctx, pullSecretName.Name, metav1.DeleteOptions{})
```

Comment:
Similar to above - do we need to delete an invalid secret if our plan is to apply unconditionally at the end? I believe any invalid fields would be overridden / replaced.

Reply:
I don't think unconditional apply will work if the user secret being of the wrong type as per the code comments.